abm/server.py

```diff
@@ -145,6 +145,24 @@ def do_get(self, context, ticket: fl.Ticket):
         # return dataset as arrow flight record batches
         return fl.GeneratorStream(schema, batches)
 
+    '''
+    Serve arrow flight do_put requests
+    '''
+    def do_put(self, context, descriptor, reader, writer):
+        asset_name = json.loads(descriptor.command)['asset']
+        logger.info('getting flight information',
+            extra={'command': descriptor.command,
+                   DataSetID: asset_name,
+                   ForUser: True})
+        with Config(self.config_path) as config:
+            df_bytes = []
+            asset_conf = config.for_asset(asset_name)
+            connector = GenericConnector(asset_conf, logger, self.workdir)
+            batches = reader.read_all().combine_chunks().to_batches(max_chunksize=1)
```

Comment:
Please explain this line to me.

Reply:
This line gets a list of `RecordBatche` objects from a Table, where each `Recordbatch` represents one row.